6_predict/src/plot_diagnostics.R

```diff
@@ -0,0 +1,265 @@
+                       grp = c("Rain","Snow","Rain+Snow","CONUS"))
+  
+  p3 <- ggplot(data = plt_df, aes(x = grp, y = perf)) +
+    geom_bar(stat="identity", width = 0.6) +
+    theme_bw() +
+    xlab('Training Region') +
+    ylab('RMSE') +
+    ggtitle('Testing Region: Snowmelt-Dominated') +
+    scale_x_discrete(limits=c("Rain","Snow","Rain+Snow","CONUS")) +
+    theme(axis.title.x = element_text(size = 20),
+          axis.title.y = element_text(size = 20),
+          axis.text.x = element_text(size = 18),
+          plot.title = element_text(size = 20)) +
+    geom_text(aes(label = round(perf,1)), vjust=-0.3, size=3.5)
+  
+  ggsave(filename = fileout[3], plot = p3, device = 'png')
+
+  return(fileout)
+}
+
+get_perf_metric <- function(model_fit, perf_metric){
+  #'
+  #' @description returns the performance metric for the fitted model
+  #'
+  #' @param model_fit fitted model
+  #' @param perf_metric performance metric name
+  #'
+  #' @return performance metric value
+  
+  model_fit$.estimate[model_fit$.metric == perf_metric]
+}
+
+plot_metric_boxplot <- function(data_split, metric, region, out_dir){
+  #'
+  #' @description returns boxplots comparing the training and testing splits
+  #' for the metric.
+  #'
+  #' @param data_split the training and testing split. Example: p6_Boruta_CONUS_g2$input_data
+  #' @param metric performance metric name
+  #' @param region the modeling region
+  #'
+  #' @return filepath to the resulting plot
+  
+  fileout <- file.path(out_dir, paste0('train_test_boxplot_', metric, '_', region, '.png'))
+  
+  png(filename = fileout, width = 4, height = 4, units = 'in', res = 200)
+  boxplot(data_split$training[[metric]],
+          data_split$testing[[metric]], 
+          names = c('Training', 'Testing'))
+  dev.off()
+  
+  return(fileout)
+}
+
+
+plot_pred_obs <- function(df_pred_obs, metric, region, out_dir,
+                          from_predict = FALSE, model_wf = NULL, pred_data = NULL){
+  #'
+  #' @description returns boxplots comparing the training and testing splits
+  #' for the metric.
+  #'
+  #' @param df_pred_obs df with obs and .pred columns
+  #' @param metric performance metric name
+  #' @param region the modeling region
+  #'
+  #' @return filepath to the resulting plot
+  
+  fileout <- file.path(out_dir, paste0('pred_obs_scatter', metric, '_', region, '.png'))
+  
+  if(from_predict){
+    #predict from provided workflow and data
+    df_pred_obs <- predict(model_wf, new_data = pred_data, type = 'numeric') %>%
+      mutate(obs = pred_data[[metric]])
+  }
+  
+  plt_lim <- max(c(df_pred_obs$obs, df_pred_obs$.pred))
+  
+  png(filename = fileout, width = 4, height = 4, units = 'in', res = 200)
+  plot(df_pred_obs$obs, df_pred_obs$.pred,
+       xlim = c(0,plt_lim), ylim = c(0,plt_lim),
+       xlab = 'Observed', ylab = 'Predicted', cex = 0.4, pch = 16)
+  lines(c(0,plt_lim), c(0,plt_lim), col = 'red')
```

Comment:
add a title with the metric name? title(metric)

Reply:
Thanks for pointing out missing metric names in titles. I'll add metric names to the titles on all of the figures